### PR TITLE
feat: Add featuregate to resourceTypes

### PR DESCRIPTION
**Pull request by Mrod1598**
### Proposed Change

1. Add featuregate to resourceTypes
2. Set seeded resources to the base gates

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: model/resource_type.go [hunk 1]
```diff
@@ -325,6 +328,20 @@ func (rt *ResourceType) evalTemplate(r ResourceTypeTemplate, nameProvider otel.C
...
+// ----------------------------------------------------------------------
+// featureGate
+// ----------------------------------------------------------------------
+
+// featureGate returns the feature flag for this resource type
+func (rt *ResourceType) FeatureGate() string {
+	return rt.Spec.FeatureGate
+}
+
+// IsFeatureEnabled returns true if the feature flag for this resource type is enabled
```
Comment (cpheps) [inline model/resource_type.go:340]: Comment is wrong

File: model/resource_type.go [hunk 2]
```diff
@@ -325,6 +328,20 @@ func (rt *ResourceType) evalTemplate(r ResourceTypeTemplate, nameProvider otel.C
 	return set
 }
 
+// ----------------------------------------------------------------------
+// featureGate
+// ----------------------------------------------------------------------
+
+// featureGate returns the feature flag for this resource type
```
Comment (cpheps) [inline model/resource_type.go:335]: nit: Comment should be capitalized like function name

File: store/store.go [hunk 3]
```diff
@@ -409,6 +409,20 @@ func GetSeedResources(ctx context.Context, files embed.FS, folders []string) ([]
...
+		switch rType := resource.(type) {
+		case *model.SourceType:
+			rType.SetFeatureGate("base-sources")
+		case *model.DestinationType:
+			rType.SetFeatureGate("base-destinations")
+		case *model.ProcessorType:
+			rType.SetFeatureGate("base-processors")
+		}
+		allEmbedded[resourceIndex] = resource
+	}
```
Comment (BinaryFissionGames) [inline store/store.go:424]: Forgot about this when we were looking over this together, but these should be "defaults" (only set if there is no feature gate).

Comment (Mrod1598) [reply store/store.go:424]: Good point, updated.

File: model/resource_type.go (hunk 2 above)
Comment (Mrod1598) [reply model/resource_type.go]: Updated.

File: model/resource_type.go (hunk 1 above)
Comment (Mrod1598) [reply model/resource_type.go]: Updated.

**Review by cpheps — APPROVED** [review]

**Review by BinaryFissionGames — APPROVED** [review]

**Pull request merged**